org.eclipse.xtext/src/org/eclipse/xtext/workspace/FileProjectConfig.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2015, 2017 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.workspace;
+
+import static org.eclipse.xtext.xbase.lib.IterableExtensions.*;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.xtend2.lib.StringConcatenation;
+import org.eclipse.xtext.util.UriUtil;
+import org.eclipse.xtext.xbase.lib.Functions.Function1;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.eclipse.xtext.xbase.lib.Pure;
+
+import com.google.common.base.Objects;
+
+public class FileProjectConfig implements IProjectConfig {
+	private final URI path;
+
+	private final String name;
+
+	private final Set<FileSourceFolder> sourceFolders = new HashSet<>();
+
+	private final IWorkspaceConfig workspaceConfig;
+
+	public FileProjectConfig(final URI path) {
+		this(path, ((IWorkspaceConfig) null));
+	}
+
+	public FileProjectConfig(final URI path, final String name) {
```

Comment:
I would try to minimize unneeded `final`s

Reply:
Also done for other files